From 87de2fc27261fa23a504f7df0fc26f16a38691c9 Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Sat, 23 Jun 2018 23:25:48 -0700
Subject: [PATCH] RetroPlayer: Fix segfault on subsequent game error

This fixes a segfault that results from initializing a game add-on twice.
---
 xbmc/cores/RetroPlayer/RetroPlayer.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/xbmc/cores/RetroPlayer/RetroPlayer.cpp b/xbmc/cores/RetroPlayer/RetroPlayer.cpp
index 380cc635d960..2b54e342b6ea 100644
--- a/xbmc/cores/RetroPlayer/RetroPlayer.cpp
+++ b/xbmc/cores/RetroPlayer/RetroPlayer.cpp
@@ -193,6 +193,8 @@ bool CRetroPlayer::OpenFile(const CFileItem& file, const CPlayerOptions& options
   }
   else
   {
+    if (m_gameClient)
+      m_gameClient->Unload();
     m_gameClient.reset();
     m_input.reset();
     m_streamManager.reset();
